code/modules/integrated_electronics/core/printer.dm

```diff
@@ -113,6 +114,12 @@
 
 	var/datum/component/material_container/materials = GetComponent(/datum/component/material_container)
 
+	//Preparing the browser
+	var/datum/browser/popup = new(user, "printernew", "Integrated Circuit Printer", 800, 630) // Set up the popup browser window
+	if(!(in_range(src, user) || issilicon(user)))
```

Comment:
Yeah no, the problem is that it closes the popup on when you are too distant. If you had no popup and would try to close it, it would give you 2 warnings and an error and rightly so. I could move it before the if(isnull(current_category)) or make the popup not close on being too far from the printer by removing popup.close() if you want to.